gradoop-flink/src/main/java/org/gradoop/flink/algorithms/gelly/randomjump/functions/VCIComputeFunction.java

```diff
@@ -0,0 +1,222 @@
+   * aggregator for visited vertices and accumulates the visited vertices from the previous
+   * superstep.
+   */
+  @Override
+  public void preSuperstep() {
+    startIndices = (List) getBroadcastSet(startIndicesBroadcastSet);
+    vertexIndices = (List) getBroadcastSet(vertexIndicesBroadcastSet);
+    visitedVerticesAggregator = getIterationAggregator(visitedVerticesAggregatorName);
+    LongValue previousAggregate = getPreviousIterationAggregate(visitedVerticesAggregatorName);
+    if (previousAggregate != null) {
+      currentVisitedCount += previousAggregate.getValue();
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * Initially starts the first walk or jump from the given start vertices. Afterwards starts a
+   * walk or jump from a vertex, if it received messages from other vertices. Stops the
+   * computation and therefor the iteration if the number of currently visited vertices exceeds
+   * the number of vertices to visit.
+   *
+   * @param vertex The vertex this computation in a superstep is running for.
+   * @param messages Iterator over all incoming messages
+   */
+  @Override
+  public void compute(Vertex<Long, VCIVertexValue> vertex, MessageIterator<NullValue> messages) {
+
+    if (currentVisitedCount < verticesToVisit) {
+      List<Edge<Long, Long>> edgesList = Lists.newArrayList(getEdges());
+      Tuple2<VCIVertexValue, Boolean> valueWithHasChanged = Tuple2.of(vertex.getValue(), false);
+      if (startIndices.contains(vertex.getId()) && !valueWithHasChanged.f0.isVisited()) {
+        valueWithHasChanged = walkToRandomNeighbor(valueWithHasChanged, edgesList);
+      } else if (messages.hasNext()) {
+        for (NullValue msg : messages) {
+          valueWithHasChanged = walkToRandomNeighbor(valueWithHasChanged, edgesList);
+        }
+      }
+      if (valueWithHasChanged.f1) {
+        setNewVertexValue(valueWithHasChanged.f0);
+      }
+    }
+  }
+
+  /**
+   * Performs a walk to a random neighbor by sending a message to a target from an unvisited
+   * outgoing edge. Sets the vertex value as visited if necessary and updates the visited edge ids.
+   * Sets a boolean flag, if the vertex value has changed.
+   * Alternatively performs a jump to a random vertex with a probability given in
+   * {@link #jumpProbability} or if there are no unvisited outgoing edges.
+   * Returns the vertex value with the boolean flag as {@code Tuple2} eventually.
+   *
+   * @param valueWithHasChanged {@code Tuple2} containing the vertex value and a boolean flag
+   *                            determining if the value has changed
+   * @param edgesList List of all outgoing edge for the vertex
+   * @return {@code Tuple2} containing the vertex value and a boolean flag determining if the
+   *         value has changed
+   */
+  private Tuple2<VCIVertexValue, Boolean> walkToRandomNeighbor(
+    Tuple2<VCIVertexValue, Boolean> valueWithHasChanged, List<Edge<Long, Long>> edgesList) {
+
```

Comment:
Again, empty lines at start of method looks weird.

Reply:
Done